test/expect-tests/dune_console/dune_console_tests.ml

```diff
@@ -1,7 +1,36 @@
 open Stdune
-
-let escape str =
-  str |> String.split_lines |> List.map ~f:String.escaped |> List.iter ~f:print_endline
+module Re = Dune_re
+
+let () =
+  Dune_tests_common.init ();
+  (* We explitily override the console width to fall back to 40. *)
+  Dune_config.Config.init
+    (String.Map.of_list_exn [ "console_width_fallback", (Loc.none, "40") ])
+;;
+
+let escape_and_split_lines str = String.split_lines str |> List.map ~f:String.escaped
+
+(* Split on new lines and then escape remaining control characters. *)
+let escape str = escape_and_split_lines str |> List.iter ~f:print_endline
+
+(* Split on new lines and then escape remaining control characters. Also print:
+   - Number of lines in [str].
+   - Number of line clears "\027[2K".
+   - Number of carriage returns.
+*)
+let escape_and_count =
+  let clear_re = Re.compile (Re.str "\027[2K") in
+  fun str ->
+    let lines = escape_and_split_lines str in
+    List.iter lines ~f:print_endline;
+    print_endline "-------";
+    Printf.printf "Line count: %d\n" (List.length lines);
+    ((* We pad the string with spaces to ensure that we don't miss any clears at the end
+        of the string. *)
+     let lines_split_by_clear = Re.split clear_re (" " ^ str ^ " ") in
```

Comment:
Weird that you need to do this. Can't you just `Re.full_split` and count the delimiters?